common/ledger/util/leveldbhelper/leveldb_helper.go

```diff
@@ -107,14 +107,16 @@ func (dbInst *DB) Get(key []byte) ([]byte, error) {
 	dbInst.mutex.RLock()
 	defer dbInst.mutex.RUnlock()
 	value, err := dbInst.db.Get(key, dbInst.readOpts)
-	if err == leveldb.ErrNotFound {
-		value = nil
-		err = nil
+	if errors.Is(err, leveldb.ErrNotFound) {
+		return nil, nil
 	}
 	if err != nil {
 		logger.Errorf("Error retrieving leveldb key [%#v]: %s", key, err)
 		return nil, errors.Wrapf(err, "error retrieving leveldb key [%#v]", key)
 	}
+	if value == nil {
+		value = []byte{}
+	}
```

Comment:
Why is this change needed? Doesn't it change the behavior in a way that may impact callers?

Reply:
Without this change, everything stops working.

Previously the goleveldb library returned []byte{}
now it returns []byte(nil)

And in the fabric returning nil it is considered that no key is found in the database
Instead of finding and fixing all the places where this behavior is used, I simulated our old behavior.

I should probably add a comment

Reply:
https://github.com/syndtr/goleveldb/commit/75fe51978d5f82f9b62d1025e1d061cb2008c38b

here's the commit that made this change

Reply:
Thanks @pfi79 , the original author @manish-sethi will review this one.

Reply:
LGTM give the change in leveldb behavior.